src/prefect/transactions.py

```diff
@@ -101,16 +104,27 @@ def __enter__(self):
             raise RuntimeError(
                 "Context already entered. Context enter calls cannot be nested."
             )
-        # set default commit behavior
+        parent = get_transaction()
+        if parent:
+            self._stored_values = copy.deepcopy(parent._stored_values)
```

Comment:
Not relevant to this PR, but the `deepcopy` here is a pretty interesting choice.  It means that a nested transaction wouldn't affect the stored values of a parent transaction, which seems right to me.  Not entirely sure it makes sense to me that a nested transaction can _see_ the parent's values in the first place, though.

Reply:
This bit is just a refactor and doesn't change any functionality. You can see https://github.com/PrefectHQ/prefect/pull/14924 where this functionality was added.

Reply:
@chrisguidry yea, just to add some motivational color: this came from a user request who was beginning to test out transactions